debezium-connector-mariadb/src/main/java/io/debezium/connector/mariadb/MariaDbConnectorTask.java

```diff
@@ -143,10 +143,15 @@ protected ChangeEventSourceCoordinator<MariaDbPartition, MariaDbOffsetContext> s
 
         LOGGER.info("Reconnecting after finishing schema recovery");
         try {
+            if (!connection.isConnected()) {
```

Comment:
I don't think this is going to work. If you look at the source code it just checks if the underlying connection was created and not explicitly closed. It does not check if the connection is healthy. For that you'd need to exectue a test statement to verify the connection is still alive and working.

Reply:
What do you prefer? connection.isValid(someTimeout) or connection.execute("SELECT 1").

Reply:
I'd say `connection.execute` - in this case we cn be really sure of end-to-end testing of the connection.